waspc/README.md

```diff
@@ -17,150 +17,135 @@ If you would like to make your first contribution, here is a handy checklist we
 
-### Build
+Running `./run` without any arguments will print help/usage, which is a good way to discover the typical `waspc` development workflows.
 
-```
-cabal build
-```
+> [!TIP]
+> To make it easy to run the `./run` script from any place in your waspc codebase, you can create a bash alias that points to it, e.g.:
+> ```sh
+> alias wrun="/home/martin/git/wasp-lang/wasp/waspc/run"
+> ```
 
-to build the library and `wasp` executable.
+### Setup
 
-This might take a while (10 mins) if you are doing it for the very first time, since `cabal` will need to download the external dependencies.
-If that is the case, relax and feel free to get yourself a cup of coffee! When somebody asks what you are doing, you can finally rightfully say "compiling!" :D.
+#### Haskell toolchain
 
-:warning: You may need to run `cabal update` before attempting to build if it has been some time since your last update.
+The best way to install Haskell toolchain is via [ghcup](https://www.haskell.org/ghcup/).
 
-:warning: If you are on Mac and get "Couldn't figure out LLVM version!" error message while building, make sure you have LLVM installed and that it is correctly exposed via env vars (PATH, LDFLAGS, CPPFLAGS). The easiest way to do it is by just running `brew install llvm@13`, this should install LLVM and also set up env vars.
+Once you have `ghcup` installed, run `ghcup tui` command to install and select correct versions of `cabal` (package manager), `hls` (language server) and `ghc` (compiler).
 
-:warning: If the LLVM error persists even after its installation, you may need to manually add it your PATH. To do this, you should add the following to end of your shell rc file (e.g. _~/.bashrc_ or _~/.zshrc_): `export PATH="/opt/homebrew/Cellar/llvm@13/13.0.1_2/bin/:$PATH"`.
+To find out what the correct versions of these tools are for the `waspc`:
+ - Run `./run ghcup-set` to check which exact versions of `ghc` and `hls` you should use.
+ - As for `cabal`, just go with the latest one.
 
-### Compiling TypeScript Packages
+> [!NOTE]
+> On Mac, we recommend using the official [ghcup](https://www.haskell.org/ghcup/) installer over Homebrew, as it works out of the box.
 
-Before running `wasp-cli`, examples, or tests locally, you must compile the bundled TypeScript packages used by the CLI.
+#### Javascript / TypeScript toolchain
 
-Wasp bundles some TypeScript packages into the installation artifact (eg: deployment scripts), which end up in the installed version's `waspc_datadir`. To do so in CI, it runs `./tools/install_packages_to_data_dir.sh`.
+You need to have the correct version of `node` (and `npm`) installed and set.
 
-For more details about TypeScript packages in Wasp, see [TypeScript Packages section](#typescript-packages).
+Check [.nvmrc](.nvmrc) file to learn which version of `node` that should be.
 
-Run this once after cloning:
+### Build
 
-```bash
-./run wasp-packages:compile
+```sh
+./run build:all
```

Comment:
Now we tell them here to run this new command. So they don't have to know too much -> it just means they are building the whole project.